detection_rules/schemas/definitions.py

```diff
@@ -28,6 +29,11 @@
 BRANCH_PATTERN = f'{VERSION_PATTERN}|^master$'
 
 NON_DATASET_PACKAGES = ['apm', 'endpoint', 'system', 'windows', 'cloud_defend', 'network_traffic']
+NON_PUBLIC_FIELDS = {
+    "related_integrations": (Version.parse('8.3.0'), None),
+    "required_fields": (Version.parse('8.3.0'), None),
+    "setup": (Version.parse('8.3.0'), None)
```

Comment:
We have this in `rule.py` as well. Will we eventually be moving everything to `non_public_fields` or are we keeping build time and non-public fields separate in terms of functionality they are referenced for?

Reply:
Good catch. Originally this was scoped just for 2931. Now that 2930 is in this as well, it actually may be better to just reuse `BUILD_FIELD_VERSIONS` for now.

Reply:
I think it makes sense to keep them separate. While it is duplicative, it may be more difficult to separate them later should it be necessary.  :+1: